coderd/database/dbauthz/dbauthz.go

```diff
@@ -3526,12 +3526,15 @@ func (q *querier) UpdateWorkspaceTTL(ctx context.Context, arg database.UpdateWor
 	return update(q.log, q.auth, fetch, q.db.UpdateWorkspaceTTL)(ctx, arg)
 }
 
-func (q *querier) UpdateWorkspacesDormantDeletingAtByTemplateID(ctx context.Context, arg database.UpdateWorkspacesDormantDeletingAtByTemplateIDParams) error {
-	fetch := func(ctx context.Context, arg database.UpdateWorkspacesDormantDeletingAtByTemplateIDParams) (database.Template, error) {
-		return q.db.GetTemplateByID(ctx, arg.TemplateID)
+func (q *querier) UpdateWorkspacesDormantDeletingAtByTemplateID(ctx context.Context, arg database.UpdateWorkspacesDormantDeletingAtByTemplateIDParams) ([]database.Workspace, error) {
+	template, err := q.db.GetTemplateByID(ctx, arg.TemplateID)
```

Comment:
Be careful that the semantics didn't change here since you're moving away from `fetchAndExec`.

Reply:
What do you mean? 🤔

Reply:
Previously the function was using `fetchAndExec` but now you've replaced that, so just take care that this didn't change the behaviour of this function.

Reply:
Now it is returning the affected workspaces, which I think is okay and is related to this method signature. Wdyt?